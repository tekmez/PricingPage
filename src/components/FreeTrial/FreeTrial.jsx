import { Text, View } from 'react-native'
import styles from './FreeTrial.style'
import CustomSwitch from '../Switch'
const FreeTrial = () => {
  return (
    <View style={styles.container}>
      <View>
        <Text testID="free-trial-light-title" style={styles.lightText}>
          not sure yet ?
        </Text>
        <Text testID="free-trial-regular-title" style={styles.regularText}>
          Enable Free Trial
        </Text>
      </View>
      <CustomSwitch />
    </View>
  )
}

export default FreeTrial
