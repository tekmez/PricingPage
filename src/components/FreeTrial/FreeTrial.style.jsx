import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    paddingVertical: 13,
    paddingHorizontal: 20,
    borderRadius: 8,
    backgroundColor: '#FFF5F5',
    borderWidth: 2,
    borderColor: '#FF0000',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  lightText: {
    fontFamily: 'Inter-Medium',
    color: '#717578',
    fontSize: 12,
    lineHeight: 16,
  },
  regularText: {
    fontFamily: 'Inter-Bold',
    fontSize: 14,
    lineHeight: 20,
  },
})

export default styles
