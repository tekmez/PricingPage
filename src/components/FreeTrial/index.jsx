export { default } from './FreeTrial.jsx'
