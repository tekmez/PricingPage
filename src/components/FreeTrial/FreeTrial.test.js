import React from 'react'
import { render } from '@testing-library/react-native'
import FreeTrial from './FreeTrial'

jest.mock('../Switch', () => {
  return jest.fn().mockImplementation(() => {
    return null
  })
})

describe('FreeTrial', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<FreeTrial />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('renders correctly', () => {
    const comp = render(<FreeTrial />)

    // Test light title
    const lightTitle = comp.getByTestId('free-trial-light-title')
    expect(lightTitle).toBeTruthy()
    expect(lightTitle.props.children).toBe('not sure yet ?')

    // Test regular title
    const regularTitle = comp.getByTestId('free-trial-regular-title')
    expect(regularTitle).toBeTruthy()
    expect(regularTitle.props.children).toBe('Enable Free Trial')
  })
})
