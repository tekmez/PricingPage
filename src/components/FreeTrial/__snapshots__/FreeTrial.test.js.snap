// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FreeTrial should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFF5F5",
      "borderColor": "#FF0000",
      "borderRadius": 8,
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingVertical": 13,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#717578",
          "fontFamily": "Inter-Medium",
          "fontSize": 12,
          "lineHeight": 16,
        }
      }
      testID="free-trial-light-title"
    >
      not sure yet ?
    </Text>
    <Text
      style={
        {
          "fontFamily": "Inter-Bold",
          "fontSize": 14,
          "lineHeight": 20,
        }
      }
      testID="free-trial-regular-title"
    >
      Enable Free Trial
    </Text>
  </View>
</View>
`;
